### PR TITLE
zh-trans: add content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md

### PR DESCRIPTION
**Pull request by donhui**
add content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md

### CODE REVIEW

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+* `network-plugin`: The network plugin to use from `cni-bin-dir`.  It must match the name reported by a plugin probed from the plugin directory.  For CNI plugins, this is simply "cni".
+-->
+## 安装
+
+kubelet 有一个单独的默认网络插件，以及一个对整个集群通用的默认网络。
+它在启动时探测插件，记住找到的内容，并在 pod 生命周期的适当时间执行所选插件（这仅适用于 Docker，因为 rkt 管理自己的 CNI 插件）。
+在使用插件时，需要记住两个 Kubelet 命令行参数：
+
+* `cni-bin-dir`： Kubelet 在启动时探测这个目录中的插件
+* `network-plugin`： 要使用的网络插件来自 `cni-bin-dir`。它必须与从插件目录探测到的插件报告的名称匹配。对于 CNI 插件，它简称 "cni"。
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md]: > 它简称 "cni"

其值为 "cni"

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+
+By default if no kubelet network plugin is specified, the `noop` plugin is used, which sets `net/bridge/bridge-nf-call-iptables=1` to ensure simple configurations (like Docker with a bridge) work correctly with the iptables proxy.
+-->
+## 网络插件要求
+
+除了提供[`NetworkPlugin` 接口](https://github.com/kubernetes/kubernetes/tree/{{< param "fullversion" >}}/pkg/kubelet/dockershim/network/plugins.go)来配置和清理 pod 网络之外，该插件还可能需要对 kube-proxy 的特定支持。
+iptables 代理显然依赖于 iptables，插件可能需要确保容器通信对 iptables 可用。
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md]:
```suggestion
iptables 代理显然依赖于 iptables，插件可能需要确保 iptables 能够监控容器的网络通信。
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+例如，如果插件将容器连接到 Linux 网桥，插件必须将 `net/bridge/bridge-nf-call-iptables` 系统参数设置为`1`，以确保 iptables 代理正常工作。
+如果插件不使用 Linux 网桥（而是类似于 Open vSwitch 获取其它一些机制），它应该确保为代理正确路由了容器流量。
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md]:
```suggestion
如果插件不使用 Linux 网桥（而是类似于 Open vSwitch 或者其它一些机制），它应该确保为代理对容器通信执行正确的路由。
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+<!--
+#### Support traffic shaping
+
+The CNI networking plugin also supports pod ingress and egress traffic shaping. You can use the official [bandwidth](https://github.com/containernetworking/plugins/tree/master/plugins/meta/bandwidth)
+plugin offered by the CNI plugin team or use your own plugin with bandwidth control functionality.
+
+If you want to enable traffic shaping support, you must add a `bandwidth` plugin to your CNI configuration file
+(default `/etc/cni/net.d`).
+-->
+#### Support 流量整形
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md]:
```suggestion
#### 支持流量整形
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+
+CNI 网络插件还支持 pod 入口和出口流量整形。
+您可以使用 CNI 插件团队提供的[宽带](https://github.com/containernetworking/plugins/tree/master/plugins/meta/bandwidth)插件，
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md]:
```suggestion
您可以使用 CNI 插件团队提供的[bandwidth](https://github.com/containernetworking/plugins/tree/master/plugins/meta/bandwidth)插件，
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+也可以使用您自己的具有带宽控制功能的插件。
+
+如果您想要启用流量整形支持，你必须将`宽带`插件添加到 CNI 配置文件
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md]:
```suggestion
如果您想要启用流量整形支持，你必须将`bandwidth`插件添加到 CNI 配置文件
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md [hunk 7]
```diff
@@ -0,0 +1,273 @@
...
+* The standard CNI `bridge`, `lo` and `host-local` plugins are required, at minimum version 0.2.0. Kubenet will first search for them in `/opt/cni/bin`. Specify `cni-bin-dir` to supply additional search path. The first found match will take effect.
+* Kubelet must be run with the `--network-plugin=kubenet` argument to enable the plugin
+* Kubelet should also be run with the `--non-masquerade-cidr=<clusterCidr>` argument to ensure traffic to IPs outside this range will use IP masquerade.
+* The node must be assigned an IP subnet through either the `--pod-cidr` kubelet command-line option or the `--allocate-node-cidrs=true --cluster-cidr=<cidr>` controller-manager command-line options.
+-->
+### kubenet
+
+Kubenet 是一个非常基本的、简单的网络插件，仅适用于 Linux。
+它本身并不实现更高级的功能，如跨节点网络或网络策略。
+它通常与云提供商一起使用，云提供商为节点之间或单节点环境中的通信设置路由规则。
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md]:
```suggestion
它通常与云驱动一起使用，云启动为节点间或单节点环境中的通信设置路由规则。
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md [hunk 8]
```diff
@@ -0,0 +1,273 @@
...
+
+Kubenet 创建了一个名为 `cbr0` 的网桥，并为每个 pod 创建了一个 veth 对，每个 pod 的主机端都连接到 `cbr0`。
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md]:
```suggestion
Kubenet 创建名为 `cbr0` 的网桥，并为每个 pod 创建了一个 veth 对，每个 pod 的主机端都连接到 `cbr0`。
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md [hunk 9]
```diff
@@ -0,0 +1,273 @@
...
+这个 veth 对的 pod 端会被分配一个 IP 地址，该 IP 地址通过配置或 controller-manager 从分配的节点范围中分配。
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md]:
```suggestion
这个 veth 对的 pod 端会被分配一个 IP 地址，该 IP 地址隶属于节点所被分配的 IP 地址范围内。节点的 IP 地址范围则通过配置或控制器管理器来设置。
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md [hunk 10]
```diff
@@ -0,0 +1,273 @@
...
+`cbr0` 被分配一个 MTU，该 MTU 匹配主机上已启用的正常接口的最小 MTU。
+
+这个插件需要做一些事情：
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md]:
```suggestion
使用此插件还需要一些其他条件：
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md [hunk 11]
```diff
@@ -0,0 +1,273 @@
...
+
+* 需要标准的 CNI `bridge`、`lo` 以及 `host-local` 插件，最低版本是0.2.0。Kubenet 首先在 `/opt/cni/bin` 中搜索它们。 指定 `cni-bin-dir` 以提供其它的搜索路径。首次找到的匹配将生效。
+* Kubelet 必须和 `--network-plugin=kubenet` 参数一起运行，才能启用该插件。
+* Kubelet 还应该和 `--non-masquerade-cidr=<clusterCidr>` 参数一起运行，以确保超出此范围的 IP 流量将使用 IP masquerade。
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md]:
```suggestion
* Kubelet 还应该和 `--non-masquerade-cidr=<clusterCidr>` 参数一起运行，以确保超出此范围的 IP 流量将使用 IP 伪装。
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md [hunk 12]
```diff
@@ -0,0 +1,273 @@
...
+* 节点必须被分配一个 IP 子网，通过 `--pod-cidr` kubelet 命令行选项或 `--allocate-node-cidrs=true --cluster-cidr=<cidr>` controller-manager 命令行选项.
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md]:
```suggestion
* 节点必须被分配一个 IP 子网，通过kubelet 命令行的 `--pod-cidr` 选项或控制器管理器的命令行选项 `--allocate-node-cidrs=true --cluster-cidr=<cidr>` 来设置。
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md [hunk 13]
```diff
@@ -0,0 +1,273 @@
...
+
+Where needed, you can specify the MTU explicitly with the `network-plugin-mtu` kubelet option.  For example,
+on AWS the `eth0` MTU is typically 9001, so you might specify `--network-plugin-mtu=9001`.  If you're using IPSEC you
+might reduce it to allow for encapsulation overhead e.g. `--network-plugin-mtu=8873`.
+
+This option is provided to the network-plugin; currently **only kubenet supports `network-plugin-mtu`**.
+-->
+### 自定义 MTU（使用 kubenet）
+
+MTU应始终正确配置，以获得最佳的网络性能。
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md]:
```suggestion
要获得最佳的网络性能，必须确保 MTU 的取值配置正确。
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md [hunk 14]
```diff
@@ -0,0 +1,273 @@
...
+网络插件通常会尝试推断出一个合理的 MTU，但有时候这个逻辑不会产生一个最优的 MTU。
+例如，如果 Docker 网桥或其他接口有一个小的 MTU, kubenet 当前将选择该 MTU。
+或者如果您正在使用 IPSEC 封装，则必须减少 MTU，并且这种计算超出了大多数网络插件的范围。
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md]:
```suggestion
或者如果您正在使用 IPSEC 封装，则必须减少 MTU，并且这种计算超出了大多数网络插件的能力范围。
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md [hunk 15]
```diff
@@ -0,0 +1,273 @@
...
+
+如果需要，您可以使用 `network-plugin-mtu` kubelet 选项显式的指定 MTU。
+例如：在 AWS 上 `eth0` MTU 通常是 9001，因此您可以指定 `--network-plugin-mtu=9001`。
+如果您正在使用 IPSEC ，您可以减少它以允许封装开销，例如 `--network-plugin-mtu=8873`。
+
+此选项提供给网络插件； 当前 **仅 kubenet 支持 `network-plugin-mtu`**。
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md]:
```suggestion
此选项会传递给网络插件； 当前 **仅 kubenet 支持 `network-plugin-mtu`**。
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md [hunk 16]
```diff
@@ -0,0 +1,273 @@
...
+<!--
+## Usage Summary
+
+* `--network-plugin=cni` specifies that we use the `cni` network plugin with actual CNI plugin binaries located in `--cni-bin-dir` (default `/opt/cni/bin`) and CNI plugin configuration located in `--cni-conf-dir` (default `/etc/cni/net.d`).
+* `--network-plugin=kubenet` specifies that we use the `kubenet` network plugin with CNI `bridge` and `host-local` plugins placed in `/opt/cni/bin` or `cni-bin-dir`.
+* `--network-plugin-mtu=9001` specifies the MTU to use, currently only used by the `kubenet` network plugin.
+-->
+## 使用总结
+
+* `--network-plugin=cni` 指定了我们使用的 `cni` 网络插件， 实际的 CNI 插件二进制位于 `--cni-bin-dir`（默认是 `/opt/cni/bin`）， CNI 插件配置位于 `--cni-conf-dir`（默认是 `/etc/cni/net.d`）。
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md]:
```suggestion
* `--network-plugin=cni` 用来表明我们要使用 `cni` 网络插件，实际的 CNI 插件可执行文件位于 `--cni-bin-dir`（默认是 `/opt/cni/bin`）下， CNI 插件配置位于 `--cni-conf-dir`（默认是 `/etc/cni/net.d`）下。
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md [hunk 17]
```diff
@@ -0,0 +1,273 @@
...
+* `--network-plugin=kubenet` 指定了我们使用的 `kubenet` 网络插件，CNI `bridge` 和 `host-local` 插件位于 `/opt/cni/bin` 或 `cni-bin-dir` 中。
```
Comment (tengqm) [inline content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md]:
```suggestion
* `--network-plugin=kubenet` 用来表明我们要使用 `kubenet` 网络插件，CNI `bridge` 和 `host-local` 插件位于 `/opt/cni/bin` 或 `cni-bin-dir` 中。
```

File: content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md (hunk 7 above)
Comment (donhui) [reply content/zh/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md]: 它通常与云驱动一起使用，云驱动为节点间或单节点环境中的通信设置路由规则。

这里，我手动修改了～

**Comment by donhui** [comment]
@tengqm Thanks for your review, all suggestions were applied.

**Pull request merged**